cmd/dep/ensure.go

```diff
@@ -127,7 +128,18 @@ func (cmd *ensureCommand) Run(ctx *dep.Ctx, args []string) error {
 	if err != nil {
 		return errors.Wrap(err, "ensure ListPackage for project")
 	}
-
+	if len(params.RootPackageTree.Packages) == 0 {
+		return errors.New("Current directory doesn't contain any go code")
+	}
+	for impPth, pckOrMap := range params.RootPackageTree.Packages {
+		if err := pckOrMap.Err; err != nil {
+			if noGoErr, ok := err.(*build.NoGoError); ok {
+				return errors.Wrap(noGoErr, "Current directory doesn't contain any go code")
```

Comment:
Sorry, I can see how my earlier review was ambiguous, but I think these returns are premature. We need to look through all the `PackageOrErrs`, and only return an err if none of them meet our conditions - as-written, won't this bail out if even a single problem is detected?